frontend/appflowy_tauri/src/appflowy_app/components/layout/NavigationPanel/FolderItem.hooks.ts

```diff
@@ -30,58 +56,130 @@ export const useFolderEvents = (folder: IFolder) => {
     setShowRenamePopup(true);
   };
 
-  const changeFolderTitle = (newTitle: string) => {
+  const changeFolderTitle = async (newTitle: string) => {
+    await FolderEventUpdateApp(
+      UpdateAppPayloadPB.fromObject({
+        name: newTitle,
+        desc: '',
+        app_id: folder.id,
+      })
+    );
     appDispatch(foldersActions.renameFolder({ id: folder.id, newTitle }));
   };
 
   const closeRenamePopup = () => {
     setShowRenamePopup(false);
   };
 
-  const deleteFolder = () => {
+  const deleteFolder = async () => {
     closePopup();
+    await FolderEventDeleteApp(
+      AppIdPB.fromObject({
+        value: folder.id,
+      })
+    );
     appDispatch(foldersActions.deleteFolder({ id: folder.id }));
```

Comment:
I think the backend service(for example, AppBackendService) should go to the foldersActions' handler. Just like the `Bloc`(Flutter) does.

Reply:
refactored to use backend service